### ISSUE TITLE
Cannot run the test example

### ISSUE THREAD

**Issue by acms-llnl** [nanoporetech/remora#136]
Hello, I am trying to use remora on my Apple Silicon M2 to try and get modified bases outside of dorado but before I get it going with my own data I was trying to get used to software by running the test example using the command lines on the GitHub page. I have tried to install remote using several different methods (conda/python environment, installed from source, from latest package and cloning the repository all followed by the recommended pip install commands). The way I got the least errors was using the vent commands from the README followed by pip install -e remora/[tests], but even then I am running into errors, I cannot seem to find a way around it (if someone posted this before, I am so sorry I couldn't find the post). See bellow for install and error messages when trying to run the test example:

## Install:
[
![Screenshot 2023-11-17 at 11 33 31 AM](https://github.com/nanoporetech/remora/assets/150087056/195fee9c-c522-493a-9f74-4b24c523d981)
](url)

## Errors:
![Screenshot 2023-11-17 at 11 27 27 AM](https://github.com/nanoporetech/remora/assets/150087056/71ccee10-6ccf-470f-8c1f-35eae17aa6f1)

Has anyone had similar problems or have tips on how to help me out? Thanks in advance!!!

**Comment by marcus1487** [comment]
TL;DR try: `MP_WAIT_TIME=5 remora [rest of command]`

There are a number of issues when running on Mac for which I have not been able to find robust workarounds. To run commands you'll have to set the `MP_WAIT_TIME` environment variable. This will wait this number of seconds after spawning a process before continuing so that the OS can write a file. Without this wait mac OS continues without a usable subprocess. Note that there are also issues with the python logging module on a mac which do not save logging messages to the log files. For all of these reasons I do most of my development work on unix.

**Comment by marcus1487** [comment]
For more info on the sleep requirement on mac see [this thread](https://stackoverflow.com/questions/67913861/python-multiprocessing-weird-behavior-when-not-using-time-sleep/67914433#67914433).

And for logging module note the [docs here](https://docs.python.org/2/howto/logging-cookbook.html#logging-to-a-single-file-from-multiple-processes) (though I'm still not sure why unix is able to handle logging from `multiprocessing.Process`es).

**Comment by acms-llnl** [comment]
Hi @marcus1487 , Thanks for the tips!! The workaround with `MP_WAIT_TIME=5 remora [rest of command]` worked for the beginning of the process! Now I am struggling with the training models... if you have any more tips I am all ears hahaha .. sorry to bother so much!! I am fairly new to all these GPU requirements and training models .. so any help is appreciated!!

I will definitely dive deep on the links you sent to understand the issues better! Thank you so much for the help.

Here is my modified command for the test run on model training... I changed the `--device mps` and `--model ~/software/remora_test/remora_latest/models/ConvLSTM_w_ref.py` because my GPU is not CUDA enabled and for whatever reason the default --model could not find the model.... so I specified it. 

`remora model train   \
train_dataset.jsn   \
--model remora/models/ConvLSTM_w_ref.py   \
 --chunk-context 50 50   \
--output-path train_results \
--device mps \
--model ~/software/remora_test/remora_latest/models/ConvLSTM_w_ref.py`

For this specific command line I get an error, with or without `MP_WAIT_TIME=5` and the error is the same. See the following screenshot: 
![Screenshot 2023-11-17 at 2 16 52 PM](https://github.com/nanoporetech/remora/assets/150087056/a6a002cc-d090-4b28-814f-49a7ae771892)


I saw in a different forum people switching to the `--device cpu` instead and seemed to have worked for them but I still get an error message. Although not the same as previously... I get an error message about **not enough chunks** (below). I understand that a bypass from GPUs to CPU would likely get it going, but I am assuming that would be a lot slower than the GPU-enabled process... and given the large datasets generated by PromethION, I am assuming the GPU would be preferable.

![Screenshot 2023-11-17 at 2 21 05 PM](https://github.com/nanoporetech/remora/assets/150087056/9eca6662-c0e6-4c9e-8469-2e450852e143)


Have you run into those issues before too? Let me know if you have any other workarounds hahaha Thank you so much for all the help!

**Comment by marcus1487** [comment]
I have not put any effort into using a mac GPU for Remora training. There may be some fundamental work required to the repo for this. We will be unlikely to support this in the medium term. I would suggest experimenting with the `--device cpu` option, and finding a standard unix+cuda environment for larger scale training.

For the second error it looks like the dataset might now have generated correctly. Could you run `remora dataset inspect` to see if the model generated correctly? Could you also take a look at the dataset prep log and see if there are any obvious errors in there?

**Comment by marcus1487** [comment]
Closing due to inactivity. Please re-open if this is still an issue.

**Comment by wenjiany** [comment]
I had the same "not enough chunks" error on a nvidia gpu.  It seems we need to provide --num-test-chunks 200 (or some other numbers smaller, the default 10000 seems too large for the test data).
